certora/specs/ghoVariableDebtToken-rayMulDiv-summarization.spec

```diff
@@ -0,0 +1,276 @@
+     *   DummyPool.sol (linked to POOL)  *;
+     *************************************/
+    function _._getReserveNormalizedVariableDebtWithBlockTimestamp(address asset, uint256 timestamp) internal => indexAtTimestamp(timestamp) expect uint256;
+    
+    /************************************;
+     *	GhoVariableDebtTokenHarness.sol	*;
+     *************************************/
+    function discStrategy.calculateDiscountRate(uint256, uint256) external returns (uint256) envfree;
+    
+    /************************************;
+     *	GhoVariableDebtTokenHarness.sol	*;
+     *************************************/
+    function getUserCurrentIndex(address) external returns (uint256) envfree;
+    function getUserDiscountRate(address) external returns (uint256) envfree;
+    function getUserAccumulatedDebtInterest(address) external returns (uint256) envfree;
+    function getBalanceOfDiscountToken(address) external returns (uint256);
+
+    /********************************;
+     *	GhoVariableDebtToken.sol	*;
+     *********************************/
+    function totalSupply() external returns(uint256) envfree;
+    function balanceOf(address) external returns (uint256);
+    function mint(address, address, uint256, uint256) external returns (bool, uint256);
+    function burn(address ,uint256 ,uint256) external returns (uint256);
+    function scaledBalanceOf(address) external returns (uint256) envfree;
+    function getBalanceFromInterest(address) external returns (uint256) envfree;
+    function rebalanceUserDiscountPercent(address) external;
+    function updateDiscountDistribution(address ,address ,uint256 ,uint256 ,uint256) external;
+}
+
+ghost rayMul_gst(mathint , mathint) returns uint256 {
+    //axiom 1==1;
+        axiom forall mathint x. forall mathint y. //rayMul_gst(x,y)+0 == x;
+      (
+       ((x==0||y==0) => rayMul_gst(x,y)==0)
+       &&
+       x <= to_mathint(rayMul_gst(x,y)) && to_mathint(rayMul_gst(x,y)) <= 2*x
+      )    ;
+}
+ghost rayDiv_gst(mathint , mathint) returns uint256 {
+    //    axiom 1==1;
+        axiom forall mathint x. forall mathint y. //rayDiv_gst(x,y)+0 == x;
+      (
+       x/2 <= to_mathint(rayDiv_gst(x,y)) && to_mathint(rayDiv_gst(x,y)) <= x
+      );
+}
+
+
+definition MAX_DISCOUNT() returns uint256 = 10000; // equals to 100% discount, in points
+
+ghost mapping(address => mapping (uint256 => uint256)) discount_ghost;
+ghost mapping(uint256 => uint256) index_ghost;
+
+/**
+* Query index_ghost for the index value at the input timestamp
+**/
+function indexAtTimestamp(uint256 timestamp) returns uint256 {
+    require index_ghost[timestamp] >= ray();
+    return index_ghost[timestamp];
+}
+
+/**
+* Query discount_ghost for the [user]'s balance of discount token at [timestamp]
+**/
+function balanceOfDiscountTokenAtTimestamp(address user, uint256 timestamp) returns uint256 {
+	return discount_ghost[user][timestamp];
+}
+
+
+function get_discount_scaled(env e, address user) returns uint256 {
```

Comment:
this is an overfit that i dislike.
Not sure what to say here. If there's another property that is more general and can catch the mutant that can be better

Reply:
I agree.
But I tried to write something like: if the index has grown, and the percentage-discount is not 0 then there should be a discount, but it doesn't hold because of rounding error.
So I'm using the same calculation that takes place in the design.